### DISCUSSION TITLE
Does Ant Media Server support RTMP playback?

### DISCUSSION THREAD

**Discussion by Mohit-3196** [ant-media/Ant-Media-Server#4874] [FAQs] [answered]
I use OBS encoder to publish an rtmp stream to Ant Media Server.

I can publish the stream easily but if I try to play the same rtmp stream URL with VLC or any other player, the rtmp can not be played.

How can I play RTMP with Ant Media Server or is there any workaround solution?

**Comment by yashtandon113** [comment, marked as answer]
Ant Media Server does not recommend RTMP stream playback because it is no longer maintained. Instead, you can use WebRTC (0.5 seconds), Dash (3-5 sec), or HLS (8-10 sec) playback options.

RTMP playback is disabled by default in v2.5.1 and above. To enable it, edit and add the below flag to the _/usr/local/antmedia/webapps/LiveApp/WEB_INF/red5-web.properties_ file.

`settings.rtmpPlaybackEnabled=true // By default, it is false.
`

After saving the changes, please restart the server with `sudo service antmedia restart`

You can also set this from the Advanced Settings on the web panel starting AMS v2.6.2.
https://antmedia.io/docs/guides/configuration-and-testing/ams-application-configuration/

Now, one should be able to play RTMP stream.

**Comment by AbdulRehman6919** [comment]
Hi @yashtandon113 
I  am having the same issue, I did exactly the same thing as mentioned above but this solution does not works for me. I also attached the screenshot of error along with the code when I am trying to get frame from the rtmp stream but every time it says **Server error: RTMP playback is disabled**

```
import cv2  # Import OpenCV lib
from PIL import Image 

stream_url = "rtmp://165.227.114.95/LiveApp/<stream_id>"
cap = cv2.VideoCapture() # Open video source as object
cap.open(stream_url)
print(cap)

print("Cap object: ", cap.isOpened())
while cap.isOpened():  # Untill end of file/error occured
    ret, frame = cap.read()  # Read frame as object - numpy.ndarray, ret is a confirmation of a successfull retrieval of the frame
    print("ret is: ", ret)
    if ret: 
        display(Image.fromarray(frame))
    else:
        break
cap.release() 
```

The error I was facing is :

![image](https://github.com/ant-media/Ant-Media-Server/assets/57502456/46145b4b-36dc-43f6-a7d6-908d8f70d9b2)

I tried to download one frame from ffmpeg command too  but it kept on saying the same error. Below is the screenshot of the error I was facing:

![image](https://github.com/ant-media/Ant-Media-Server/assets/57502456/5cb31bf3-e6ef-47e3-8d0a-68520e37aa96)

**Reply by USAMAWIZARD** [reply]
@AbdulRehman6919 Did you restarted the server after applying the settings ?
